packages/flutter/lib/src/widgets/two_dimensional_viewport.dart

```diff
@@ -894,9 +892,112 @@ abstract class RenderTwoDimensionalViewport extends RenderBox implements RenderA
   }
 
   @override
-  RevealedOffset getOffsetToReveal(RenderObject target, double alignment, { Rect? rect }) {
-    // TODO(Piinks): Add this back in follow up change (ensureVisible), https://github.com/flutter/flutter/issues/126299
-    return const RevealedOffset(offset: 0.0, rect: Rect.zero);
+  RevealedOffset getOffsetToReveal(
+    RenderObject target,
+    double alignment, {
+    Rect? rect,
+    AxisDirection? axisDirection,
+  }) {
+    // We must know which axis we are revealing for.
+    assert(axisDirection != null);
+
+    // Starting at `target` and walking towards the root:
+    //  - `child` will be the last object before we reach this viewport, and
+    //  - `pivot` will be the last RenderBox before we reach this viewport.
+    RenderObject child = target;
+    RenderBox? pivot;
+    // Find the direct child of the viewport that contains our target
+    while (child.parent != this) {
+      final RenderObject parent = child.parent! as RenderObject;
+      if (child is RenderBox) {
+        pivot = child;
+      }
+      child = parent;
+    }
+    final double offset = switch (axisDirectionToAxis(axisDirection!)) {
+      Axis.vertical => verticalOffset.pixels,
+      Axis.horizontal => horizontalOffset.pixels,
+    };
+
+    // `rect` in the new intermediate coordinate system.
+    final Rect rectLocal;
+    // Our new reference frame render object's main axis extent.
+    final double pivotExtent;
+    if (pivot != null) {
+      assert(pivot.parent != null);
+      assert(pivot.parent != this);
```

Comment:
The comment on line 906 said that pivot is the last RenderBox before the ViewPort. Doesn't that mean that its parent is `this` Viewport? In 1D scrollables there are slivers between the Viewport and the last RenderBox, what do we have in the 2d case there where we don't have slivers?

What am I missing about how this code works?

Reply:
Now that I am re-reading the comment on lone 905/906: In a 2d viewport, what is the difference between `child` and `pivot` ? Wouldn't they always point to the same?

Reply:
In the case of the code in the PR description, after 7 loops the child is the repaint boundary and pivot is the actual 200x200 constrained box it wraps. The `target` the method is called with is the focus node of the actual checkbox within, so we bubble up from there to the top level child.